### PR TITLE
Packaging UPI playbooks for download - OSDOCS1665

### PR DESCRIPTION
**Pull request by maxwelldb**
Context: https://issues.redhat.com/browse/OSDOCS-1665

This PR instructs users to fetch Ansible playbooks from installer release branches directly. Previously, they would copy + paste these playbooks from the docs.

### CODE REVIEW

**Comment by maxwelldb** [comment]
TODO: Swap {playbook-version} out for {product-version} immediately prior to merge. Former is just for easy review + testing.

**Comment by maxwelldb** [comment]
Any thoughts, @pierreprinetti ? 

Impacted assemblies:
- UPI: https://playbook-packaging-osdocs1665--ocpdocs.netlify.app/openshift-enterprise/latest/installing/installing_openstack/installing-openstack-user.html
- UPI w/Kuryr: https://playbook-packaging-osdocs1665--ocpdocs.netlify.app/openshift-enterprise/latest/installing/installing_openstack/installing-openstack-user-kuryr.html
- UPI cluster removal: https://playbook-packaging-osdocs1665--ocpdocs.netlify.app/openshift-enterprise/latest/installing/installing_openstack/uninstalling-openstack-user.html

**Review by pierreprinetti — APPROVED** [review]
👍

**Comment by maxwelldb** [comment]
Thanks!

**Comment by pierreprinetti** [comment]
Just one nit, and I'm not sure how you want to deal with it:

In the "uninstall" phase, the download-script downloads again "inventory.yaml". This means that the user will override the modifications they made during the "install" phase, which might be a problem.

I'd remove "inventory.yaml" from the "uninstall" download-script, because they will most likely have to reuse the same inventory they modified in  the "install" phase

**Comment by maxwelldb** [comment]
@pierreprinetti That's fair. Since I'd gather that we prefer users keep the modified inventory.yaml file around, I propose:

1. Adding a note to the installation flows that encourages the user to preserve inventory.yaml after installation. 
2. Adding a prerequisite to have the inventory.yaml from installation in order to remove a UPI cluster. 
3. Optional for 4.7, but nice: Adding instructions for people who want to remove a UPI cluster, but do not have the original inventory.yaml. Not sure how complicated this could get.

Thoughts?

**Comment by pierreprinetti** [comment]
> Adding instructions for people who want to remove a UPI cluster, but do not have the original inventory.yaml.

I expect this not to be possible.

## My rationale:

The UPI playbooks (that is, the files you download with the `curl` command) are useless in their original form, as by design, they mimic an IPI install.

_If you plan on using the UPI playbooks as-they-are, you should probably just run the IPI installer instead.
Corollary:
**In a typical UPI install, both inventory & playbooks are either modified, or heavily modified.**_

The real value of the UPI playbooks is that the user changes them to adapt them to their infrastructure.

* First, they'll address the little differences by changing the variables we provide in `inventory.yaml`.
* Then, they will probably remove or adapt parts of the scripts themselves to address the big things.

**I would also expect the user to match any change they make to the `up` playbooks to the `down-` playbooks.**

Ultimately, only the customised `inventory.yaml` and `down-` playbooks will be able to destroy the cluster.

**Comment by maxwelldb** [comment]
@pierreprinetti Yeah, I didn't expect a simple way to do this. :) 

Strike out item 3, then. https://github.com/openshift/openshift-docs/pull/27912/commits/1763d960f3d253c5f954b08453d70de0902c3f3e just does the first two, anyhow. 

It sounds like we should add a reminder to retain _all_ of the install playbooks so that the uninstall playbooks can be updated prior to use, then.

**Comment by maxwelldb** [comment]
@pierreprinetti Feel free to +1/-1 the previews in https://github.com/openshift/openshift-docs/pull/27912#issuecomment-740212636. They should show the latest in ~15m.

**Comment by maxwelldb** [comment]
TODO: Revise language in installation mods to emphasize that you're using the same playbooks throughout the process--you're not starting over half-way through with a new inventory.yaml.

**Comment by pierreprinetti** [comment]
Here's one thought. I don't know if it's worth addressing in your patch; I just want you to be aware of it.

The user workflow that I expect (read: how I do things) is:
1. I make a change in a "up" playbook
2. I immediately mirror the change it in the corresponding `down-` playbook
1. I run the "up" playbooks
1. They inexorably fail
1. I run the `down-` suite to reset the target environment (this works because I adapted them on step 2)
1. back to step 1

For this workflow to work, I need the "up" playbooks and the `down-` playbooks right from the start.

With the current "create" and "destroy" sections, the user might start adapting their own "up" playbooks before being made aware that `down-` playbooks exist.

**Comment by maxwelldb** [comment]
@pierreprinetti That seems like how it'll be in the end. 

(I've been trying to approach some modular documentation guidelines that don't work very well here.)

**Comment by maxwelldb** [comment]
Another pass:

-     UPI: https://playbook-packaging-osdocs1665--ocpdocs.netlify.app/openshift-enterprise/latest/installing/installing_openstack/installing-openstack-user.html
-     UPI w/Kuryr: https://playbook-packaging-osdocs1665--ocpdocs.netlify.app/openshift-enterprise/latest/installing/installing_openstack/installing-openstack-user-kuryr.html
-     UPI cluster removal: https://playbook-packaging-osdocs1665--ocpdocs.netlify.app/openshift-enterprise/latest/installing/installing_openstack/uninstalling-openstack-user.html

Previews should be updated within the hour.

**Review by pierreprinetti — APPROVED** [review]
Looks good!

**Comment by wjiangjay** [comment]
LGTM

**Comment by maxwelldb** [comment]
Rebase done. Will squash post-review. 

@openshift/team-documentation PTAL when you have a moment. Thanks!

**Comment by maxwelldb** [comment]
TODO post-review: update version attribute for prod. Currently, I have it set up to facilitate review.

File: modules/installation-osp-downloading-playbooks.adoc [hunk 1]
```diff
@@ -0,0 +1,50 @@
+// Module included in the following assemblies:
+// * installing/installing_openstack/installing-openstack-installer-user.adoc
+// * installing/installing_openstack/installing-openstack-installer-user-kuryr.adoc
+:playbook-version: 4.7
```
Comment (maxwelldb) [inline modules/installation-osp-downloading-playbooks.adoc:4]: FIXME immediately prior to merging.

File: modules/installation-osp-downloading-playbooks.adoc [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+* The curl command-line tool is available on your machine.
+
+.Procedure
+
+. To download the playbooks to your working directory, run the following script from a command line:
++
+[source,terminal,subs=attributes+]
+----
+$ xargs -n 1 curl -O <<< '
```
Comment (maxwelldb) [inline modules/installation-osp-downloading-playbooks.adoc:21]: FIXME immediately prior to merging; playbook-version is for review only.

File: modules/installation-osp-deleting-bootstrap-resources.adoc [hunk 3]
```diff
@@ -4,30 +4,22 @@
 // * installing/installing_openstack/installing-openstack-kuryr.adoc
```
Comment (lbarbeevargas) [inline modules/installation-osp-deleting-bootstrap-resources.adoc:4]: Should this assembly be `installing-openstack-user-kuryr.adoc`? Just noticed it when I tried getting to the preview for this module.

File: modules/installation-osp-downloading-playbooks.adoc [hunk 4]
```diff
@@ -0,0 +1,50 @@
+// Module included in the following assemblies:
+// * installing/installing_openstack/installing-openstack-installer-user.adoc
+// * installing/installing_openstack/installing-openstack-installer-user-kuryr.adoc
```
Comment (lbarbeevargas) [inline modules/installation-osp-downloading-playbooks.adoc:3]: I think these two assemblies are:
installing/installing_openstack/installing-openstack-user.adoc
installing/installing_openstack/installing-openstack-user-kuryr.adoc

File: modules/installation-uninstall-infra.adoc [hunk 5]
```diff
@@ -3,109 +3,21 @@
 // * installing/installing_osp/uninstalling-cluster-osp.adoc
```
Comment (lbarbeevargas) [inline modules/installation-uninstall-infra.adoc:3]: I think this assembly file path needs to be updated. `installing/installing_openstack/uninstalling-openstack-user.adoc`

**Review by lbarbeevargas — COMMENTED** [review]
This looks really great! Just a few comments and some assembly references that need to be updated.

File: modules/installation-osp-deleting-bootstrap-resources.adoc (hunk 3 above)
Comment (maxwelldb) [reply modules/installation-osp-deleting-bootstrap-resources.adoc:4]: Yeah, whoops.

**Comment by maxwelldb** [comment]
Verifying on 4.7 d.o.c. before cherrypicking to 4.6.

**Comment by maxwelldb** [comment]
Welp, and there's that. Will manually CP later assuming it all looks good on d.o.c.

**Comment by maxwelldb** [comment]
Seems fine.

**Pull request merged**